app/routes/application.js

```diff
@@ -0,0 +1,33 @@
+import Route from '@ember/routing/route';
+import ENV from 'website-my/config/environment';
+
+const API_BASE_URL = ENV.BASE_API_URL;
+export default class ApplicationRoute extends Route {
+  model = async () => {
+    const defaultName = 'User';
+    const defaultPicture = 'dummyProfilePicture.png';
+
+    try {
+      const response = await fetch(`${API_BASE_URL}/users/self`, {
+        credentials: 'include',
+      });
+      const userData = await response.json();
+      if (response.status === 200 && !userData.incompleteUserDetails) {
+        const firstName = userData.first_name;
+        const profilePictureURL = userData.picture.url;
+        return (
+          { firstName, profilePictureURL } ?? { defaultName, defaultPicture }
+        );
```

Comment:
I'm not sure of the `??` check, what if there's a `firstName` but no `profilePictureURL`? Does this check work in that case?

Reply:
Ok will look into it and let you know

Reply:
I tested this in my local and came to know that this operator don't work with objects.

Reply:
Yep. It's not supposed to work

Reply:
Yeah